### PR TITLE
Add slippry@1.4.0 w/ git auto-update config

### LINKED ISSUE #9484: [Request] Add slippry
**Library name: slippry** 
**Git repository url: https://github.com/booncon/slippry**
**npm package url(optional):** https://www.npmjs.com/package/slippry
**License(s): MIT**
**Official homepage: http://slippry.com/**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by extend1994**
Pull request for issue: #9484 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/booncon/slippry
 * Official website (optional, not the repository): http://slippry.com/
 * NPM package url (optional): https://www.npmjs.com/package/slippry
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 404
   - Count of watchers: 42
   - Count of forks: 127
 * NPM download stats (optional):
   - Downloads in the last day: 12
   - Downloads in the last week: 31
   - Downloads in the last month: 104

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #9484, cc @booncon

### CODE REVIEW

**Comment by extend1994** [comment]
@PeterDaveHello please help me review this pull request, thank you!

**Comment by extend1994** [comment]
@PeterDaveHello I include `slippry.js` from `src` now, please help me review it, thank you.

**Comment by extend1994** [comment]
@PeterDaveHello 
Then whether 
`"fileMap": [
      {
        "basePath": "(dist|src)",
        "files": [
          "*.+(css|js)"
        ]
      }
]
`
is correct?

**Comment by PeterDaveHello** [comment]
```js
"fileMap": [
  {
    "basePath": "dist",
    "files": [
      "**/*"
    ]
  },
  {
    "basePath": "src",
    "files": [
      "*.+(css|js|map)"
    ]
  }
]
```

**Comment by extend1994** [comment]
@PeterDaveHello I updated the `filemap` in the `package.json`, please help me review it, thank you!

**Comment by extend1994** [comment]
@PeterDaveHello I updated the file structure, please help me review it, thank you.

**Comment by extend1994** [comment]
@PeterDaveHello I think that I fixed the permission problem, please help me review it, thank you.

**Comment by extend1994** [comment]
@PeterDaveHello I think that I fixed the permission problem, please help me review it, thank you.

**Comment by PeterDaveHello** [comment]
Looks like you'll need to add `images` folder: https://github.com/booncon/slippry/tree/v1.4.0/images

Ref:
https://github.com/booncon/slippry/blob/v1.4.0/dist/slippry.css#L63
https://github.com/booncon/slippry/blob/v1.4.0/dist/slippry.css#L186

**Comment by extend1994** [comment]
@PeterDaveHello Could you please tell me that adding them manually or using `fileMap/basepath` in `package.json` to get it is correct? Thank you.

**Comment by PeterDaveHello** [comment]
Of course you'll need to add them into auto-update config.

**Comment by extend1994** [comment]
@PeterDaveHello How about `files` rule in auto-update config? Is the rule like `**/*` to include all files in the folder correct? I would like to do this in case of that the author changes the files' name or content in css files.

**Comment by PeterDaveHello** [comment]
I didn't see your update about the `images` directory, for `src` folder, you can use `slippry*.+(css|js|map)`

**Comment by extend1994** [comment]
@PeterDaveHello I updated the `package.json`, please help me view it, thank you.

**Comment by extend1994** [comment]
@maruilian11 I added the images files, please help review it, thanks.

File: ajax/libs/slippry/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "*.+(css|js|map)"
+        ]
+      },
+      {
+        "basePath": "images",
```
Comment (pvnr0082t) [inline ajax/libs/slippry/package.json]: It's not good to use `images` as basePath.

Comment (extend1994) [inline ajax/libs/slippry/package.json]: @pvnr0082t In this case, I set dist and src as basePath, the best way I can think of is also setting images as basePath to include the files under images directory. Could you give me some advice about it? Thank you.

Comment (maruilian11) [inline ajax/libs/slippry/package.json]:
```
"basePath": "",
"files": [
  "images/*.+(svg|gif)"
]
```

**Comment by extend1994** [comment]
@pvnr0082t @maruilian11 please help review pull this pull request, thank you very much.

**Comment by extend1994** [comment]
@PeterDaveHello @maruilian11 @pvnr0082t please help me review this pull request again, thank you.

File: ajax/libs/slippry/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      },
+      {
+        "basePath": "src",
+        "files": [
+          "*.+(css|js|map)"
```
Comment (extend1994) [inline ajax/libs/slippry/package.json]: @maruilian11 minify tool is https://github.com/PeterDaveHello/web-minify-helper.
I don't get you about source map url?

Could you please help review it again, thank you.

**Comment by maruilian11** [comment]
`slippry.js`'s permission is wrong.

**Comment by extend1994** [comment]
@maruilian11 sorry for my carelessness, I modified the file permission, please help me review it again, thank you!

**Comment by maruilian11** [comment]
please add `.1.2` to .gitignore since it's same as `1.2`.

**Comment by extend1994** [comment]
@sufuf3 Though the file structure is different, the files the library needs can be fetched by the current auto-update config. So I think we don't need another commit. What do you think?

**Comment by maruilian11** [comment]
@extend1994 you can add it in the same commit, remember to explain it in commit msg.
Don't need to add those versions.

**Comment by extend1994** [comment]
@maruilian11 @sufuf3 I revised the commit message, please help me review this pull request, thank you.

**Comment by maruilian11** [comment]
add it here https://github.com/cdnjs/cdnjs/blob/master/.gitignore

**Comment by extend1994** [comment]
@maruilian11 @sufuf3 I updated the file and commit message, please help me review it again, thank you.

File: .gitignore [hunk 3]
```diff
@@ -191,6 +191,7 @@ node_modules/
 /ajax/libs/jquery-easing/1.3.0/
 /ajax/libs/jquery-easing/1.3.1/
 /ajax/libs/libphonenumber/libphonenumber*/
+/ajax/libs/slippry/v.1.2/
```
Comment (maruilian11) [inline .gitignore]: Not `v.1.2`, you should test it before sending the commit.

Comment (extend1994) [inline .gitignore]: @maruilian11 Is not v.1.2?
ss

Comment (PeterDaveHello) [inline .gitignore]: `/ajax/libs/slippry/v.1.2/` looks good to me, what's the problem?

Comment (maruilian11) [inline .gitignore]: @extend1994 Sorry about my mistake, I use old version of `autoupdate` to fetch the version.
I think this PR is fine now, could you rebase this branch with our latest master? Thanks.

Comment (PeterDaveHello) [inline .gitignore]: @maruilian11 I think you can help rebase in this situation, I'll rebase it this time.

Comment (PeterDaveHello) [inline .gitignore]: @maruilian11 and then you should approve it first, shouldn't you?

Comment (maruilian11) [inline .gitignore]: Because I want to approve after rebasing. Next time, I will help rebase.

**Pull request merged**